### PR TITLE
Make dependence on number of openings explicit in security estimator

### PR DESCRIPTION
**Pull request by Al-Kindi-0**
As the title says and it corrects a few minor mistakes. The security estimates do not change (when rounding to bits) after testing with some of the examples.

### CODE REVIEW

**Review by irakliyk — COMMENTED** [review]
Looks good! Thank you for updating this!

I think it would be good to add a test for `get_proven_security()` function at a few sample parameter sets. For example, we could use:`base_field_bits = 64`, `collision_resistance = 128`, `trace_domain_size = 2^20`, and then vary `num_queries`, `blowup_factor`, and `extension_degree` to get 96 bits or 128 bits of security.

These could be good "sanity checks" to make sure the numbers we are getting are as expected.

File: air/src/proof/mod.rs [hunk 1]
```diff
@@ -404,3 +406,259 @@ pub fn ceil(value: f64) -> f64 {
...
+    fn get_96_bits_security() {
+        let field_extension = FieldExtension::Cubic;
+        let base_field_bits = BaseElement::MODULUS_BITS;
+        let fri_folding_factor = 8;
+        let fri_remainder_max_degree = 127;
+        let grinding_factor = 20;
+        let mut blowup_factor = 4;
+        let mut num_queries = 80;
+        let collision_resistance = 128;
+        let trace_length = 1 << 18;
```
Comment (irakliyk) [inline air/src/proof/mod.rs:426]: minor nit: should we use `2_usize.pow(18)` to make the operation a bit more explicit?

File: air/src/proof/mod.rs [hunk 2]
```diff
@@ -404,3 +406,259 @@ pub fn ceil(value: f64) -> f64 {
...
+            fri_remainder_max_degree as usize,
+        );
+        let security_1 =
+            get_proven_security(&options, base_field_bits, trace_length, collision_resistance);
+
+        assert_eq!(security_1, 97);
+
+        // increasing the blowup factor should increase the bits of security gained per query
+        blowup_factor <<= 1;
+        num_queries -= 27;
```
Comment (irakliyk) [inline air/src/proof/mod.rs:443]: nit: I would probably redefine the variables here. For example:
```Rust
let blowup_factor = 8;
let num_queries = 53;
```

File: air/src/proof/mod.rs [hunk 3]
```diff
@@ -404,3 +406,259 @@ pub fn ceil(value: f64) -> f64 {
...
+    fn get_128_bits_security() {
+        let field_extension = FieldExtension::Cubic;
+        let base_field_bits = BaseElement::MODULUS_BITS;
+        let fri_folding_factor = 8;
+        let fri_remainder_max_degree = 127;
+        let grinding_factor = 20;
+        let mut blowup_factor = 8;
+        let mut num_queries = 85;
+        let collision_resistance = 128;
+        let trace_length = 1 << 18;
```
Comment (irakliyk) [inline air/src/proof/mod.rs:468]: Similar comment as above here and for tests below as well.

File: air/src/proof/mod.rs [hunk 4]
```diff
@@ -404,3 +406,259 @@ pub fn ceil(value: f64) -> f64 {
...
+            fri_remainder_max_degree as usize,
+        );
+        let security_1 =
+            get_proven_security(&options, base_field_bits, trace_length, collision_resistance);
+
+        assert_eq!(security_1, 128);
+
+        // increasing the blowup factor should increase the bits of security gained per query
+        blowup_factor <<= 1;
+        num_queries -= 20;
```
Comment (irakliyk) [inline air/src/proof/mod.rs:485]: Similar comments as above here and for tests below.

File: air/src/proof/mod.rs [hunk 5]
```diff
@@ -404,3 +406,259 @@ pub fn ceil(value: f64) -> f64 {
...
+            blowup_factor,
+            grinding_factor,
+            field_extension,
+            fri_folding_factor as usize,
+            fri_remainder_max_degree as usize,
+        );
+        let security_1 =
+            get_proven_security(&options, base_field_bits, trace_length, collision_resistance);
+
+        assert_eq!(security_1, 67);
```
Comment (irakliyk) [inline air/src/proof/mod.rs:523]: This is the highest security level we can get from quadratic extension field (without using unique decoding regime), right?

**Review by irakliyk — APPROVED** [review]
The tests look great! Thank you! I left a few small nits inline - but this is basically ready to merge.

File: air/src/proof/mod.rs (hunk 1 above)
Comment (Al-Kindi-0) [reply air/src/proof/mod.rs]: Done

File: air/src/proof/mod.rs (hunk 5 above)
Comment (Al-Kindi-0) [reply air/src/proof/mod.rs:525]: Yes indeed, for the specified trace length and blowup factor

**Review by irakliyk — APPROVED** [review]
All looks good! Thank you!

**Pull request merged**